src/main/java/org/opensearch/geospatial/plugin/GeospatialPlugin.java

```diff
@@ -81,6 +113,11 @@ public Collection<Object> createComponents(
         IndexNameExpressionResolver indexNameExpressionResolver,
         Supplier<RepositoriesService> repositoriesServiceSupplier
     ) {
+        // Initialize DatasourceUpdateRunner
+        DatasourceRunner.getJobRunnerInstance().setClient(client);
+        DatasourceRunner.getJobRunnerInstance().setClusterService(clusterService);
+        DatasourceRunner.getJobRunnerInstance().setThreadPool(threadPool);
+
         return List.of(UploadStats.getInstance());
```

Comment:
idea of create components function is to create all the objects that this plugin needs, I see that we are not returning DataSourceRunner instance at the end why that is the case?

Reply:
You can refer how to write job scheduler extension here. https://github.com/opensearch-project/job-scheduler/tree/main/sample-extension-plugin

Job scheduler does not utilize DI pattern to get extensions. So, no use of returning the object here.